runtime/compiler/env/j9method.cpp

```diff
@@ -7397,14 +7397,14 @@ TR_J9MethodParameterIterator::TR_J9MethodParameterIterator(TR_J9MethodBase &j9Me
 
 TR::DataType TR_J9MethodParameterIterator::getDataType()
    {
-   if (*_sig == 'L' || *_sig == '[')
+   if (*_sig == 'L' || *_sig == '[' || *_sig == 'Q') //not sure if this is right
       {
       _nextIncrBy = 0;
       while (_sig[_nextIncrBy] == '[')
     {
     ++_nextIncrBy;
     }
-      if (_sig[_nextIncrBy] != 'L')
+   if (_sig[_nextIncrBy] != 'L' || _sig[_nextIncrBy] != 'Q')
```

Comment:
This should be `if (_sig[_nextIncrBy] != 'L' && _sig[_nextIncrBy] != 'Q')` - using `||` the condition would always be true.